astropy/table/tests/test_init_table.py

```diff
@@ -505,13 +505,16 @@ def test_init_and_ref_from_dict(table_type, copy):
 
 @pytest.mark.usefixtures('table_type')
 def test_init_from_row_OrderedDict(table_type):
-        row1 = OrderedDict([('b', 1), ('a', 0)])
-        row2 = OrderedDict([('b', 11), ('a', 10)])
-        rows12 = [row1, row2]
-        row3 = dict([('b', 1), ('a', 0)])
-        row4 = dict([('b', 11), ('a', 10)])
-        rows34 = [row3, row4]
-        t1 = table_type(rows=rows12)
-        t2 = table_type(rows=rows34)
-        assert t1.colnames == ['b', 'a']
-        assert t2.colnames == ['a', 'b']
+    row1 = OrderedDict([('b', 1), ('a', 0)])
+    row2 = {'a': 10, 'b': 20}
```

Comment:
Note here that I changed `row2` a little here to adapt from @taldcroft 's example as I don't want to have too much duplicate code that basically tests for the same stuff.